test/withDataProviders.test.js

```diff
@@ -0,0 +1,145 @@
+/* global test, expect, beforeEach, describe */
+/* eslint-disable no-console */
+import React from 'react'
+import ReactDOM from 'react-dom'
+import {withDataProviders} from '../src/withDataProviders'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {newTestApp, getData, getDataWithCount, GET_DATA_DELAY} from './common'
+import {Promise} from 'bluebird'
+
+beforeEach(() => {
+  getDataWithCount.counter = 0
+})
+
+test('withDataProviders (needed=true) renders child component when the data is fetched', async () => {
```

Comment:
As you require node compatible with await-async (i guess that's 8+) at least put it in a comment or readme.